tools/building.py

```diff
@@ -823,7 +823,10 @@ def metadce(js_file, wasm_file, minify_whitespace, debug_info):
   import_name_map = {}
   for item in graph:
     if 'import' in item:
-      import_name_map[item['name']] = 'emcc$import$' + item['import'][1]
+      name = item['import'][1]
+      import_name_map[item['name']] = 'emcc$import$' + name
+      if asmjs_mangle(name) in settings.SIDE_MODULE_IMPORTS:
```

Comment:
Why do we root side module imports? If we don't use such an import can't we drop it?

Reply:
The side module requires them .. so we can't delete them.  Think JS library functions that are only used by the side module.  

For example, imagine that both the main module and the side module call `my_js_func`.   without this code, DCE, might notice that the main module usage can be removed and therefore try to also completely from `my_js_func`.

Historically this has always been the case since we were running with EXPORT_ALL which prevents any DCE of JS functions since they are all exported on the Module.

Reply:
In case this isn't clear `SIDE_MODULE_IMPORTS` means "symbols imported/required by the side module(s)"

Reply:
Oh right, sorry... I read the docs for `SIDE_MODULE_EXPORTS` and not `_IMPORTS`...